optaplanner-docs/src/main/asciidoc/ConstraintStreams/ConstraintStreams-chapter.adoc

```diff
@@ -57,3 +57,86 @@ and supports <<explainingTheScore,justifications>>:
                         Talk::getDurationInMinutes);
     }
 ----
+
+[[constraintStreamsGroupingAndCollectors]]
+== Grouping and Collectors
+
+In the constraint streams framework, grouping is a mechanism to group items in a stream according to user-provider
+criteria, similar to what a `GROUP BY` SQL clause does. Additionally, some grouping operations also accept one or more
+`Collector`s, which provide various aggregation functions.
+
+For example, the following code snippet will first group all processes by the computer they run on, sum up all the
+power required by the processes on that computer using the `ConstraintCollectors.sum(...)` collector, and finally
+penalize every computer whose processes consume more power than is available.
+
+[source,java,options="nowrap"]
+----
+    private Constraint requiredCpuPowerTotal(ConstraintFactory constraintFactory) {
+        return constraintFactory.from(CloudProcess.class)
+                .groupBy(CloudProcess::getComputer, sum(CloudProcess::getRequiredCpuPower))
+                .penalize("requiredCpuPowerTotal",
+                        HardSoftScore.ONE_HARD,
+                        (computer, requiredCpuPower) -> requiredCpuPower - computer.getCpuPower());
+    }
+----
+
+OptaPlanner provides several collector implementations you can use in your grouping constraint streams. This chapter
+discusses the most important ones. For a complete list, refer to the
+`org.optaplanner.core.api.score.stream.ConstraintCollectors` class.
+
+You can provide your own collectors by implementing the
+`org.optaplanner.core.api.score.stream.uni.UniConstraintCollector` interface or their `Bi...`, `Tri...` counterparts.
+The following sections will focus on the collector implementations provided out of the box.
+
+=== Counting collectors
```

Comment:
code style for product-docs friendliness: All Words in a Title Must Start with a Capital unless They Are In-between-words.
=> "Counting Collectors"

Counting != count()
I'd avoid the proliferation of additional ways of saying the same thing, it creates confusion.
Can we rename these sections to:
- Collecting `count()`
- Collecting `countDistinct()`
- Collecting `sum()`
etc?
Or just
- `count()`
- `countDistinct()`
- `sum()`

Reply:
Every title must have a unique ID, usually camel case of the title, sometimes prefixed with the parent section's title

[collectorsCount]
=== `count`


[countingCollectors]
=== Counting collectors